lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -1119,8 +1112,12 @@ struct Deduper {
   /// We insert all NLAs to the beginning of this block.
   Block *nlaBlock;
 
-  // This maps an NLA to the operations and ports that uses it.
-  DenseMap<Attribute, std::vector<AnnoTarget>> targetMap;
+  // This maps an NLA  to the operations and ports that uses it.
+  DenseMap<Attribute, llvm::SmallDenseSet<AnnoTarget>> targetMap;
```

Comment:
@dtzSiFive Any idea what the best data structure for a `map` from `key -> set` is?  Being a `DenseSet` with `SmallVectors` in it seems like it would get pretty giant.  There is also `SmallSet` which grows to being a `std::set` instead of a `DenseSet`.  Not sure what other options are out there.